### DISCUSSION TITLE
Encrypted Client Hello Question

### DISCUSSION THREAD

**Discussion by Romulo-Romero** [DNSCrypt/dnscrypt-proxy#2614] [Q&A]
Hello all,
I have been using dnscrypt configured with ODOH servers, and now chromium with ECH enabled, so I can actually enjoy the anonymity of using ODOH added with the privacy of ECH, but I have one question:
When I open wireshark and log traffic, I can actually see the SNI unencrypted, is this expected ? It seems to me that for some reason it isn't working properly otherwise I wouldn't be able to see SNI unencrypted, am I right?
I tested in two different websites to check if ESNI is working, and in both sites shows as it is using ECH or ESNI.
Can anyone confirm to me if the wireshark capture is the expected behavior or not ?
Thanks

**Comment by jedisct1** [comment]
For ECH to work, the websites you visit must support ECH as well. And the vast majority of websites don't.

**Reply by Romulo-Romero** [reply]
Hi Jedisct1,
Thank you for replying. As I understand ECH is enabled for all websites hosted in cloudflare server, correct me if I'm wrong. So I've tried cloudflare itself and other websites that checks for ECH and other cloudflare hosted websites, none of them showed as encrypted client hello in my wireshark captures. I am thinking maybe its because it sits at a different layer where the ssl communication are already decrypted in wireshark. So that doesn't really answer my question as to why SNI is shown without being encrypted even visiting cloudflare itself where I can check if ECH is working.
